### PR TITLE
feat: remove deprecated code

### PR DESCRIPTION
**Pull request by connor-baer**
Addresses #469.

## Purpose

Remove all components and other code that were deprecated in [v1.2](https://github.com/sumup-oss/circuit-ui/compare/v1.1.0...v1.2.0) (https://github.com/sumup-oss/circuit-ui/pull/470).

## Approach and changes

- The RestrictedInput has been removed.
- The State component has been removed. Use React's [`useState`](https://reactjs.org/docs/hooks-reference.html#usestate) hook instead.
- The RadioButton and Switch components no longer accept the `onToggle` prop. Use the `onChange` prop instead.
- The Input component no longer accepts `*ClassName` props. Emotion 10 uses style objects instead of class names. Use the `*Styles` props instead.
- The Heading, SubHeading, Text, and Input components no longer accept the `element` prop. Emotion 10 introduced the ability to change the HTML element. Use the `as` prop instead.
- The SvgButton component has been renamed to IconButton.
- The SideNav component has been removed. Use the Sidebar component instead.
- The Picture component has been removed.
- The MaskedInput component has been removed. Use [`react-text-mask`](https://www.npmjs.com/package/react-text-mask) directly instead.
- The Markdown component has been removed. Use [`markdown-to-jsx`](https://github.com/probablyup/markdown-to-jsx) directly instead.
- The CreditCardDetails, CardNumberInput, NameOnCardInput, SecurityCodeInput, ExpiryDateInput, and the credit card utils have been removed. Use [SumUp's card widget](https://developer.sumup.com/docs/widgets-card-v2) instead.
- The ListView component has been renamed to CardList.
- The GlobalStyles component has been renamed to BaseStyles. It no longer accepts a `custom` prop, use Emotion's [Global](https://emotion.sh/docs/globals) component instead.

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

**Comment by larimaza** [comment]
giphy

**Comment by herberthenrique** [comment]
200w_d

File: src/components/CurrencyInput/components/SimpleCurrencyInput/SimpleCurrencyInput.js [hunk 1]
```diff
@@ -93,7 +94,11 @@ const SimpleCurrencyInput = ({
   numberMask,
   ...props
 }) => (
-  <MaskedInput
+  <TextMaskInput
+    guide={false}
+    render={(ref, { defaultValue, ...renderProps }) => (
+      <Input value={defaultValue} {...renderProps} deepRef={ref} />
+    )}
```
Comment (hleote) [inline src/components/CurrencyInput/components/SimpleCurrencyInput/SimpleCurrencyInput.js]: Nice component!! 
Didn't knew this one... can be quite useful 🙂 👍

Comment (connor-baer) [reply src/components/CurrencyInput/components/SimpleCurrencyInput/SimpleCurrencyInput.js]: Well, it’s gone from Circuit UI now 😆

**Comment by connor-baer** [comment]
@hleote Yes, we always push updates to snapshot files. They’re also checked in CI and since the pipeline ran successfully for this PR, I’m guessing it’s something on your local machine. 

Did you update your node modules? Which version of Node are you using?

File: src/components/CurrencyInput/components/SimpleCurrencyInput/SimpleCurrencyInput.js (hunk 1 above)
Comment (connor-baer) [reply src/components/CurrencyInput/components/SimpleCurrencyInput/SimpleCurrencyInput.js]: Of course, you can use [`react-text-mask`](https://www.npmjs.com/package/react-text-mask) directly instead.

**Comment by hleote** [comment]
> @hleote Yes, we always push updates to snapshot files. They’re also checked in CI and since the pipeline ran successfully for this PR, I’m guessing it’s something on your local machine.
> 
> Did you update your node modules? Which version of Node are you using?

Using Node version 12.13.0.
Ok I will check it out it's probably something with my machine in that case 👍 🙂

**Pull request merged**